### DISCUSSION TITLE
password protection

### DISCUSSION THREAD

**Discussion by SebastiaanOsinga** [DirectoryLister/DirectoryLister#695] [Ideas & Features]
having some/main directory/ies password protected so only certain users can access it.

**Comment by ciscam** [comment]
You could use your webserver.

https://httpd.apache.org/docs/current/programs/htpasswd.html
https://httpd.apache.org/docs/current/mod/mod_authn_core.html
https://nginx.org/en/docs/http/ngx_http_auth_basic_module.html

**Comment by PHLAK** [comment]
This is not something we'll be adding to Directory Lister (see #131, #206, #402). As @ciscam said, this can be implemented at the web server level.

**Comment by asbruff** [comment]
I don't want to bother or annoy anyone... It's absolutely fine that the maintainer doesn't want to implement such a feature, but the htaccess method on apache doesn't appear to be working.

Could it be simply because the script appends "?dir=" before the destination directory url?

Do I have to guess that htaccess can't be implemented in sub directories?

**Reply by PHLAK** [reply]
Due to the changes in [v5.0](https://github.com/DirectoryLister/DirectoryLister/releases/tag/5.0.0) (specifically the introduction of the `FILE_PATH` variable for listing files outside of the installation path), the web server does not handle "sub directories" any more. All traffic goes through the `index.php` file and therefore if a user has access to the main app they will have access to all files listed as well.

However, as long as you're on [v5.1](https://github.com/DirectoryLister/DirectoryLister/releases/tag/5.1.0) or later I believe you can work around this by configuring the folder you wish to be protected as a "[direct link](https://docs.directorylister.com/configuration/configuration-reference#direct_links)" and adding the `.htaccess` to that folder as usual.

For example. if you would like to password protect a folder at `protected/path`, you can add the following environment variable.

```env
DIRECT_LINKS=protected/path/**
```

In theory this should cause all files within `protected/path` to be directly linked and follow the rules of your web server and, as long as you have the `.htaccess` file there, should prompt for authentication.

**Reply by PHLAK** [reply]
On another note, I've recently started using [Authelia](https://www.authelia.com) and found the [Proxy Authorization](https://www.authelia.com/reference/guides/proxy-authorization/) feature handy for adding authorization (controlled by Authelia) to an instance of Directory Lister behind a reverse proxy. I have no expectation that this should be the recommended solution for authorization, however, I may add some documentation for those interested in the future.
